components/general/NetPyNECoordsRange.js

```diff
@@ -0,0 +1,115 @@
+var PythonControlledAdapterComponent = PythonControlledCapability.createPythonControlledControl(AdapterComponent);
+ 
+export default class NetPyNECoordsRange extends Component {
+ 
+  constructor(props) {
+    super(props);
+    this.state = {
+      currentName: props.name,
+      rangeType: null,
+    };
+    this.ranges = [
+      { value: this.props.items[0].value, stateVariable: 'rangeType' }, 
+      { value: this.props.items[1].value, stateVariable: 'rangeType' },
+    ];
+  };
+ 
+  componentDidUpdate(prevProps, prevState) {
+    if (this.state.currentName != prevState.currentName) {
+      this.updateLayout();
+    };
+  };
+ 
+  componentDidMount() {
+    this.updateLayout();
+  };
+ 
+  componentWillReceiveProps(nextProps) {
+    if (this.state.currentName != nextProps.name) {
+      this.setState({ model: nextProps.name});
+    }
+  }
+
+  updateLayout() {
+    var message = this.props.model + "['" + this.state.currentName + "']"
+    if (this.props.conds!=undefined) message = message + "['" + this.props.conds + "']";
+    const getRange = (value, stateVariable) => {
+      Utils
+        .sendPythonMessage("'" + value + "' in " + message)
+        .then((response) => {
+          if (response) {
+            var newState = {};
+            newState[stateVariable] = value;
+            this.setState(newState)
+          };
+        });
+    };
+    this.ranges.forEach((range) => { getRange(range.value, range.stateVariable) })
+  };
+     
+  createMenuItems = () => {
+    return this.props.items.map((obj) => (
+      <MenuItem
+        key={obj.value}
+        value={obj.value}
+        primaryText={obj.label}
+      />
+    ));
+  };
+   
+  render() {
+    if (this.props.conds!=undefined) {
+      var meta = this.props.model + "." + this.props.conds + "." + this.props.items[0].value;
+      var path = this.props.model + "['" + this.state.currentName + "']['" + this.props.conds + "']['" + this.state.rangeType + "']";
+    } else {
+      var meta = this.props.model + '.' + this.props.items[0].value;
+      var path = this.props.model + "['" + this.state.currentName + "']['" + this.state.rangeType + "']"
+    };
+    
+    return (
+      <div>
+        <NetPyNEField id={meta} >
+          <SelectField
+            floatingLabelText={"Range type"}
+            value={this.state.rangeType}
+            onChange={(event, index, value) => this.setState({ rangeType: value })}
+          >
+            {this.createMenuItems()}
+          </SelectField>
+        </NetPyNEField>
+        {(this.state.rangeType != undefined) ?
+          <div className={"netpyneRightField"}>
+            <PythonControlledAdapterComponent
+              model={path}
+              convertToPython={(state) => {
+                if(state[state.lastUpdated].toString().endsWith("."))
+                  return undefined;
+                if ((state.min != undefined && !isNaN(parseFloat(state.min))) && (state.max != undefined && !isNaN(parseFloat(state.max)))) {
```

Comment:
Y think we can change this line to:

`if (!isNaN(parseFloat(state.min)) && !isNaN(parseFloat(state.max)))` {